File: 400/496 - Next Greater Element I.java
import java.util.HashMap;

class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        int i, j, num;
        boolean flag;
        int[] ret = new int[nums1.length];
        HashMap<Integer, Integer> hm = new HashMap<Integer, Integer>();
        
        for (i = 0; i < nums2.length; ++i)
            hm.put(nums2[i], i);
        
        for (i = 0; i < nums1.length; ++i) {
            num = nums1[i];
            flag = true;
            
            for (j = hm.get(num) + 1; j < nums2.length; ++j) {
                if (nums2[j] > num) {
                    flag = false;
                    ret[i] = nums2[j];
                    break;
                }
            }
            
            if (flag)
                ret[i] = -1;
        }
        
        return ret;
    }
}

File: csharp/496.next-greater-element-i.cs
/*
 * @lc app=leetcode id=496 lang=csharp
 *
 * [496] Next Greater Element I
 */

// @lc code=start
public class Solution {
    public int[] NextGreaterElement(int[] nums1, int[] nums2) {
        int m = nums1.Length, n = nums2.Length;
        int[] ret = new int[m];
        for (int i = 0; i < m; ++i) {
            int num = nums1[i], j;
            for (j = 0; j < n && nums2[j] != num; ++j);
            for (j++; j < n && nums2[j] <= num; ++j);
            ret[i] = j == n ? -1 : nums2[j];
        }
        return ret;
    }
}
// @lc code=end

